src/controllers/csv-processor.ts

```diff
@@ -250,3 +315,96 @@ export const processCSV = async (dataset: Dataset, page: number, size: number):
         };
     }
 };
+
+export const processCSVFromBlobStorage = async (
+    dataset: Dataset,
+    page: number,
+    size: number
+): Promise<ViewErrDTO | ViewDTO> => {
+    const blobStoageService = new BlobStorageService();
+    try {
+        const datafiles = await dataset.datafiles;
+        const datafile: Datafile | undefined = datafiles
+            .filter((filterfile: Datafile) => filterfile.draft === true)
+            .sort(
+                (first: Datafile, second: Datafile) =>
+                    new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+            )
+            .shift();
+        if (datafile === undefined || datafile === null) {
+            return {
+                success: false,
+                errors: [
+                    {
+                        field: 'dataset',
+                        message: [
+                            { lang: ENGLISH, message: t('errors.no_datafile', { lng: ENGLISH }) },
+                            { lang: WELSH, message: t('errors.no_datafile', { lng: WELSH }) }
+                        ],
+                        tag: { name: 'erorors.no_datafile', params: {} }
+                    }
+                ],
+                dataset_id: dataset.id
+            };
+        }
+        const buff = await blobStoageService.readFile(`${datafile.id}.csv`);
+
+        const dataArray: Array<Array<string>> = (await parse(buff, {
+            delimiter: ','
+        }).toArray()) as string[][];
+        const csvheaders = dataArray.shift();
+        const total_pages = Math.ceil(dataArray.length / size);
+        const errors = validateParams(page, total_pages, size);
+        if (errors.length > 0) {
+            return {
+                success: false,
+                errors,
+                dataset_id: dataset.id
+            };
+        }
+
+        const csvdata = paginate(dataArray, page, size);
```

Comment:
The pagination stuff needs to evolve to work with seekable streams rather than fully materialised arrays.

Reply:
Possibly... We only need this code for the publisher journey though and only when we're directly handling the uploaded CSV files.  As soon as we're able to load this data in to the online cube much of this goes away as we'll just do SQL queries to get pages of the data.